src/problem_data.jl

```diff
@@ -0,0 +1,484 @@
+        "Cannot add a row with $nz indices but $(length(rval)) non-zeros"
+    ))
+
+    # Increment row counter
+    pb.ncon += 1
+    push!(pb.con_names, name)
+    push!(pb.lcon, l)
+    push!(pb.ucon, u)
+
+    if nz == 0
+        # emtpy row
+        push!(pb.arows, Row{T}(Int[], T[]))
+        return pb.ncon
+    end
+
+    # TODO: check coefficients values, e.g.
+    #   * all coefficients are finite
+    #   * remove hard zeros
+    #   * combine dupplicate indices
+    #   * check if indices are already issorted
+
+    # Create new row
+    if issorted
+        row = Row{T}(copy(rind), copy(rval))
+    else
+        # Sort indices first
+        p = sortperm(rind)
+        row = Row{T}(rind[p], rval[p])
+    end
+    push!(pb.arows, row)
+
+    # Update column coefficients
+    for (j, v) in zip(rind, rval)
+        if !iszero(v)
+            push!(pb.acols[j].nzind, pb.ncon)
+            push!(pb.acols[j].nzval, v)
+        end
+    end
+
+    # Done
+    return pb.ncon
+end
+
+"""
+    add_variable!(pb, cind, cval, obj, l, u, [name])
+
+Add one variable to the problem.
+
+# Arguments
+* `pb::ProblemData{T}`: the problem to which the new column is added
+* `cind::Vector{Int}`: row indices in the new column
+* `cval::Vector{T}`: non-zero values in the new column
+* `obj::T`: objective coefficient
+* `l::T`: column lower bound
+* `u::T`: column upper bound
+* `name::String`: column name (defaults to `""`)
+* `issorted::Bool`: indicates whether the column indices are already issorted.
+"""
+function add_variable!(pb::ProblemData{T},
+    cind::Vector{Int}, cval::Vector{T},
+    obj::T, l::T, u::T,
+    name::String="";
+    issorted::Bool=false
+)::Int where {T}
+    # Sanity checks
+    nz = length(cind)
+    nz == length(cval) || throw(DimensionMismatch(
+        "Cannot add a column with $nz indices but $(length(cval)) non-zeros"
+    ))
+
+    # Increment column counter
+    pb.nvar += 1
+    push!(pb.var_names, name)
+    push!(pb.lvar, l)
+    push!(pb.uvar, u)
+    push!(pb.obj, obj)
+
+    if nz == 0
+        push!(pb.acols, Col{T}(Int[], T[]))
+        return pb.nvar  # empty column
+    end
+
+    # TODO: check coefficients
+
+    # Create a new column
+    if issorted
+        col = Col{T}(copy(cind), copy(cval))
+    else
+        # Sort indices
+        p = sortperm(cind)
+        col = Col{T}(cind[p], cind[p])
+    end
+    push!(pb.acols, col)
+
+    # Update row coefficients
+    for (i, v) in zip(cind, cval)
+        if !iszero(v)
+            push!(pb.arows[i].nzind, pb.nvar)
+            push!(pb.arows[i].nzval, v)
+        end
+    end
+
+    # Done
+    return pb.nvar
+end
+
+"""
+    load_problem!(pb, )
+
+Load entire problem.
+"""
+function load_problem!(pb::ProblemData{T},
+    name::String,
+    objsense::Bool, obj::Vector{T}, obj0::T,
+    A::SparseMatrixCSC,
+    lcon::Vector{T}, ucon::Vector{T},
+    lvar::Vector{T}, uvar::Vector{T},
+    con_names::Vector{String}, var_names::Vector{String}
+) where {T}
```

Comment:
We can remove all the build/modify functions except this one, since we won't be handling the modelling phase.

Reply:
Done (had to keep empty!, too, without changing interface).